zetaclient/chains/evm/observer/v2_inbound.go

```diff
@@ -90,10 +90,20 @@ func (ob *Observer) observeGatewayDeposit(
 
 		msg := ob.newDepositInboundVote(event)
 
-		ob.Logger().Inbound.Info().
-			Msgf("ObserveGateway: Deposit inbound detected on chain %d tx %s block %d from %s value %s message %s",
-				ob.Chain().
-					ChainId, event.Raw.TxHash.Hex(), event.Raw.BlockNumber, event.Sender.Hex(), event.Amount.String(), hex.EncodeToString(event.Payload))
+		// skip early observed inbound that is not eligible for fast confirmation
+		if msg.ConfirmationMode == types.ConfirmationMode_FAST {
+			eligible, err := ob.IsInboundEligibleForFastConfirmation(ctx, &msg)
```

Comment:
Why do we check IsInboundEligibleForFastConfirmation when ConfirmationMode is true?

ConfirmationMode should be set at the inbound creation when IsInboundEligibleForFastConfirmation returns true

Reply:
The reason is that we need to skip posting vote if a `msg` carries `FAST` mode while not eligible for fast confirmation.